sprocs/check_course_instance_access_rule.sql

```diff
@@ -43,14 +43,16 @@ BEGIN
         -- we have an explicit institution restriction
 
         IF course_instance_access_rule.institution = 'LTI' THEN
+            available := FALSE;
+
             -- get the uid row from users
             SELECT *
             INTO user_result
             FROM users
             WHERE users.uid = check_course_instance_access_rule.uid;
 
-            IF user_result.lti_course_instance_id != course_instance_access_rule.course_instance_id THEN
-                available := FALSE;
+            IF user_result.lti_course_instance_id = course_instance_access_rule.course_instance_id THEN
+                available := TRUE;
```

Comment:
I think this will override any earlier failed checks. For example, suppose one of the `date` checks failed and set `available = FALSE`. Then we get here and set it `TRUE` again.

Was the original bug caused by regular users having `user_result.lti_course_instance_id = NULL`, which won't trigger the `!=` comparison? If so, I think an alternative fix would be to replace `!=` with `IS DISTINCT FROM`.